packages/mobile/src/verify/VerificationInputScreen.tsx

```diff
@@ -1,23 +1,181 @@
+
+  state: State = {
+    codeInputValues: [],
+    isModalVisible: false,
+  }
+
+  onChangeInputCode = (index: number) => {
+    return (value: string) => {
+      this.setState(dotProp.set(this.state, `codeInputValues.${index}`, value))
+    }
+  }
+
+  onPressCodesNotReceived = () => {
+    this.setState({ isModalVisible: true })
+  }
+
+  onPressWaitForCodes = () => {
+    this.setState({ isModalVisible: false })
+  }
+
+  onPressSkip = () => {
+    // TODO cancel verification here
+    navigate(Screens.WalletHome)
+  }
+
+  shouldShowClipboard = (value: string) => {
+    return !!extractAttestationCodeFromMessage(value)
+  }
 
   render() {
-    // const { t } = this.props
+    const { codeInputValues, isModalVisible } = this.state
+    const { t } = this.props
+
     return (
       <SafeAreaView style={styles.container}>
-        <ScrollView contentContainerStyle={styles.scrollContainer}>
+        <ScrollView
+          contentContainerStyle={styles.scrollContainer}
+          keyboardShouldPersistTaps={'always'}
+        >
           <DevSkipButton nextScreen={Screens.WalletHome} />
+          <View style={styles.iconContainer}>
+            <SmsCeloSwap width={152} height={48} />
+          </View>
+          <Text style={fontStyles.h1} testID="VerificationInputHeader">
+            {t('input.header')}
+          </Text>
+          <Text style={fontStyles.body}>
+            <Text style={fontStyles.bold}>{t('input.body1')}</Text>
+            {t('input.body2')}
+          </Text>
+          <Text style={styles.codeHeader}>{t('input.codeHeader1')}</Text>
+          <CodeInput
+            value={codeInputValues[0]}
+            placeholder={'<#> m9oASm/3g7aZ...'}
+            shouldShowClipboard={this.shouldShowClipboard}
+            onChangeText={this.onChangeInputCode(0)}
+            style={styles.codeInput}
+          />
+          <Text style={styles.codeHeader}>{t('input.codeHeader2')}</Text>
+          <CodeInput
+            value={codeInputValues[1]}
+            placeholder={'<#> m9oASm/3g7aZ...'}
+            shouldShowClipboard={this.shouldShowClipboard}
+            onChangeText={this.onChangeInputCode(1)}
+            style={styles.codeInput}
+          />
+          <Text style={styles.codeHeader}>{t('input.codeHeader3')}</Text>
+          <CodeInput
+            value={codeInputValues[2]}
+            placeholder={'<#> m9oASm/3g7aZ...'}
+            shouldShowClipboard={this.shouldShowClipboard}
+            onChangeText={this.onChangeInputCode(2)}
+            style={styles.codeInput}
+          />
+          <Link style={styles.missingCodesLink} onPress={this.onPressCodesNotReceived}>
+            {t('input.codesMissing')}
+          </Link>
         </ScrollView>
+        <Modal isVisible={isModalVisible}>
+          <View style={styles.modalContainer}>
+            <View style={styles.modalTimerContainer}>
+              <LoadingSpinner />
+              <Text style={fontStyles.body}>{'0:49'}</Text>
```

Comment:
I assume the real value will be added in the next PR?

Reply:
Yes, already done in my latest